android/src/main/kotlin/media/bcc/bccm_player/Downloader.kt

```diff
@@ -0,0 +1,171 @@
+package media.bcc.bccm_player
+
+import android.content.Context
+import android.net.Uri
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.media3.common.MediaItem
+import androidx.media3.database.StandaloneDatabaseProvider
+import androidx.media3.datasource.DefaultDataSourceFactory
+import androidx.media3.datasource.DefaultHttpDataSource
+import androidx.media3.datasource.cache.NoOpCacheEvictor
+import androidx.media3.datasource.cache.SimpleCache
+import androidx.media3.exoplayer.DefaultRenderersFactory
+import androidx.media3.exoplayer.offline.Download
+import androidx.media3.exoplayer.offline.DownloadHelper
+import androidx.media3.exoplayer.offline.DownloadManager
+import androidx.media3.exoplayer.offline.DownloadRequest
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.parcelableCreator
+import media.bcc.bccm_player.pigeon.DownloaderApi
+import media.bcc.bccm_player.pigeon.PlaybackPlatformApi
+import java.io.File
+import java.io.IOException
+import java.util.LinkedList
+import java.util.UUID
+import java.util.concurrent.Executor
+
+
+private const val DOWNLOAD_FOLDER_NAME = "downloads"
+
+var cache: SimpleCache? = null
+
+fun createDownloadManager(context: Context): DownloadManager {
+    val databaseProvider = StandaloneDatabaseProvider(context)
+    val cacheDir = File(context.filesDir, DOWNLOAD_FOLDER_NAME)
+    if (cache == null) {
+        cache = SimpleCache(cacheDir, NoOpCacheEvictor(), databaseProvider)
+    }
+    val dataSourceFactory = DefaultHttpDataSource.Factory()
+    val downloadExecutor = Executor(Runnable::run)
+    return DownloadManager(context, databaseProvider, cache!!, dataSourceFactory, downloadExecutor)
+}
+
+@Parcelize
+data class DownloadInfo(
+    val title: String,
+    val additionalData: Map<String, String>
+) : Parcelable
+
+object ParcelMarshall {
```

Comment:
Useful. Kinda crazy something like this isn't included in the lib together with parcelize.
Maybe put it in its own file